doofinder.php

```diff
@@ -779,11 +796,17 @@ private function configureHookCommon($params = false)
      */
     public function hookHeader($params)
     {
-        $this->configureHookCommon($params);
-        if (Configuration::get('DF_ENABLED_V9')) {
-            return $this->displayScriptLiveLayer();
-        } else {
-            return $this->displayScriptV7();
+        $displayMobile = $this->getShowLayerMobileConfig();
+        $displayDesktop = $this->getShowLayerConfig();
+        $isMobile = Context::getContext()->isMobile();
+        if ((!empty($isMobile) && $displayMobile) ||
+        (empty($isMobile) && $displayDesktop)) {
```

Comment:
```suggestion
        if (getDisplayLayer())
```
Maybe it's something confusing. Better to create a function that contains the logic of whether to show the layer, and in the IF here just call that function.
In addition, the function can be used for other cases